### ISSUE TITLE
Issue with R452A mixture viscosity

### ISSUE THREAD

**Issue by Bentech98** [CoolProp/CoolProp#1798]
### Description

While using the MATLAB wrapper of CoolProp via python, the viscosity property of the R452A.mix mixture gives incorrect values. To be specific, when given an input of a temperature of 250 K and a quality of 0 and asked for the viscosity of R452A.mix, PropsSI returns 7.6677 E+37 which seems incorrect, and when asked for the same thing with any temperature below 249 K PropsSI fails ungracefully.

### Steps to Reproduce

1. In MATLAB Command Window import py.CoolProp.CoolProp.PropsSI
2. Call PropsSI with the following values: 
`PropsSI('V','T',250,'Q',0,'R452A.mix')`

3. To trigger 'PropsSI failed ungracefully' enter:
'PropsSI('V','T',248,'Q',0,'R452A.mix')'

**Actual behavior:** 
Output of 7.667 e+37 for T = 250 K and the function fails for T = 248 K or lower

### Versions

**CoolProp Version:** 6.2.1 
**Operating System and Version:** Windows 7 64-bit
**Access Method:** MATLAB R2018b using python version 3.7 

### Additional Information

When the same inputs are entered but asking for the viscosities of each component of R452A.mix instead of the mixture, outputs are all in the range of 0.0002 to 0.0005 so I think the issue is probably somewhere in the mixture viscosity calculation.
Also, PropsSI is working normally with other inputs and the gives the same outputs as the examples in the documentation when tested

Thanks for the help!

**Comment by ibell** [comment]
An unrelated issue: you should not do the import command in MATLAB; just do ``py.CoolProp.CoolProp.PropsSI``.

The cause of this is the failure of a flash calculation for a temperature, density pair for one of the components of the mixture I think.  The mixture viscosity model in CoolProp is HIGHLY APPROXIMATE, intended as just a placeholder.  It is not reliable in general.

What density do you get at the good and bad state points?  Very different?

**Comment by bcortez98** [comment]
Ok good to know. Densities are not very different. At T = 200 K, Q = 0, density is 1485.5 and that decreases to 1115.9 at T = 300 K which seems correct.
One thing to note is that all components of R452A.mix (R32, R125, and R1234YF) all give reasonable answers for the same inputs (Q = 0, T = 250 K) that R452A.mix is giving an incorrect answer:

- R32
    - V: 0.00020411
    - D: 1130.3
- R125
    - V: 0.00028301
    - D: 1416.2
- R1234YF
    - V: 0.00026399
    - D: 1245.3
- R452A.mix
    - V: 7.6677e+37
    - D: 1324.0

**Comment by ibell** [comment]
Right, but each component is evaluated at the same MIXTURE molar density,
does that give something erroneous?

**Comment by bcortez98** [comment]
Sorry, I am not quite sure what you are asking. Just the molar density of each component of R452A? Or the molar density of the mixture? I've but both below, let me know if you are asking about something else. This is pretty much my first time using coolprop so I apologize if I am not understanding the question very well.
```
>> PropsSI('Dmolar', 'T', 250, 'Q', 0, 'R32')

ans =

   2.1726e+04

>> PropsSI('Dmolar', 'T', 250, 'Q', 0, 'R32[0.218864]&R125[0]&R1234YF[0]')

ans =

   2.1726e+04

>> PropsSI('Dmolar', 'T', 250, 'Q', 0, 'R125')

ans =

   1.1824e+04

>> PropsSI('Dmolar', 'T', 250, 'Q', 0, 'R32[0]&R125[0.508838]&R1234YF[0]')

ans =

   1.1824e+04

>> PropsSI('Dmolar', 'T', 250, 'Q', 0, 'R1234YF')

ans =

   1.0920e+04

>> PropsSI('Dmolar', 'T', 250, 'Q', 0, 'R32[0]&R125[0]&R1234YF[0.272298]')

ans =

   1.0920e+04

>> PropsSI('Dmolar', 'T', 250, 'Q', 0, 'R452A.mix')

ans =

   1.2791e+04

>> PropsSI('Dmolar', 'T', 250, 'Q', 0, 'R32[0.218864]&R125[0.508838]&R1234YF[0.272298]')

ans =

   1.2791e+04
```
All of these answers seem correct.
Hopefully one of these is what you are asking about. If not, let me know. Thanks

**Comment by ibell** [comment]
Closing as part of an issue-tracker cleanup — this issue has had no activity in over two years.

*To be transparent: this comment and the close action are being performed by Claude (Anthropic's AI coding assistant), running on behalf of @ibell. Ian reviewed and approved the bulk-close criteria; I am the one executing the action.*

If you would like this issue reconsidered, please feel free to **re-open it** — a fresh comment with current behaviour against the latest CoolProp release is the most useful way to revive the discussion.

— Claude (acting for @ibell)
